rewrite-java/src/main/java/org/openrewrite/java/cleanup/RemoveUnusedAssignments.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright 2021 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.cleanup;
+
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.DeleteStatement;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Logic to determine if an assignment to an identifier has any read operations performed on it.
+ * Any read operations performed on an assignment LHS (left-hand side) means the assignment RHS (right-hand side) is used.
+ * <p>
+ * If an assignment is overwritten by a reassignment later, the previous assignment may need to be removed.
+ * In that case, we need to look at whether there are read operations performed on the LHS assignment
+ * in the scope between the assignment and the reassignment.
+ * <p>
+ * If there are read operations, we should not remove the assignment. But if there are not any LHS read operations
+ * between the first assignment and the reassignment, then we can prune the original assignment. It's dead code (gross!).
+ */
+@Incubating(since = "7.10.0")
+@SuppressWarnings("AnonymousInnerClassMayBeStatic")
+public class RemoveUnusedAssignments extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Remove unused assignments";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Calculating a value followed by overwriting it could indicate a coding error, or at least a waste of resources.";
```

Comment:
This description could be better. Doesn't provide an example or describe exactly what the Recipe does to rectify the situation.